src/lib.rs

```diff
@@ -0,0 +1,458 @@
+                    ScopeSelection::Workspace => workspace_members,
+                    ScopeSelection::DefaultMembers => {
+                        // Deviating from cargo because Metadata doesn't have default members
+                        let resolve = meta.resolve.as_ref().expect("no-deps is unsupported");
+                        match &resolve.root {
+                            Some(root) => {
+                                let mut base_ids = HashSet::new();
+                                base_ids.insert(root);
+                                base_ids
+                            }
+                            None => workspace_members,
+                        }
+                    }
+                };
+
+                packages
+                    .iter()
+                    .filter(|p| !excluded_packages.contains(&meta[p].name))
+                    .copied()
+                    .collect()
+            }
+            ScopeMode::AllowList(patterns) => {
+                meta.packages
+                    .iter()
+                    // Deviating from cargo by not supporting patterns
+                    // Deviating from cargo by only checking workspace members
+                    .filter(|p| workspace_members.contains(&p.id) && patterns.contains(&p.name))
+                    .map(|p| &p.id)
+                    .collect()
+            }
+        };
+
+        meta.packages
+            .iter()
+            .filter(|&p| base_ids.contains(&p.id))
+            .collect()
+    }
+}
+
+impl Check {
+    pub fn new(current: Rustdoc) -> Self {
+        Self {
+            scope: Scope::default(),
+            current,
+            baseline: Rustdoc::from_latest_version(),
+            log_level: Default::default(),
+        }
+    }
+
+    pub fn with_package_selection(&mut self, selection: PackageSelection) -> &mut Self {
+        self.scope.mode = ScopeMode::DenyList(selection);
+        self
+    }
+
+    pub fn with_packages(&mut self, packages: Vec<String>) -> &mut Self {
+        self.scope.mode = ScopeMode::AllowList(packages);
+        self
+    }
+
+    pub fn with_baseline(&mut self, baseline: Rustdoc) -> &mut Self {
+        self.baseline = baseline;
+        self
+    }
+
+    pub fn with_log_level(&mut self, log_level: log::Level) -> &mut Self {
+        self.log_level = Some(log_level);
+        self
+    }
+
+    fn get_target_dir(&self, source: &RustdocSource) -> anyhow::Result<PathBuf> {
+        Ok(if let Some(path) = get_target_dir_from_config() {
+            path
+        } else if let Some(path) = get_target_dir_from_project_root(source)? {
+            path
+        } else if let Some(path) = get_target_dir_from_project_root(&self.current.source)? {
+            path
+        } else if let Some(path) = get_target_dir_from_project_root(&self.baseline.source)? {
+            path
+        } else {
+            std::env::current_dir()?.join("target")
+        })
+    }
```

Comment:
Now that I think about this -- the logic behind this function is not obvious at all, I should document this.

Reply:
Done in 2449acd377ed1792127a10c0ffed211094664d1b.